rust/src/action.rs

```diff
@@ -395,6 +388,20 @@ impl From<&Field> for ColumnValueStat {
     }
 }
 
+impl From<&Field> for ColumnCountStat {
+    fn from(field: &Field) -> Self {
+        match field {
+            Field::Group(group) => ColumnCountStat::Column(HashMap::from_iter(
+                group
+                    .get_column_iter()
+                    .map(|(field_name, field)| (field_name.clone(), field.into())),
+            )),
+            Field::Long(value) => ColumnCountStat::Value(*value),
+            _ => panic!("nullCount columns stats must be a struct of int64s."),
```

Comment:
I completely agree that we should not crash if the stats can't be parsed. I thought the exception would be handled though in the [calling function](https://github.com/delta-io/delta-rs/pull/714/files/bc02b13ac05388347d2ea0a233d8a88f2ed4d84b#diff-9fb70b66d3f77e15e99399b95228d124db4bd0d898e1e159bf106ccced488332R354). This might be exposing my severely lacking rust knowledge.

Reply:
agree that it would be better to use TryForm here and let the caller handle the error. In general, we strive to be a panic free library :)

Reply:
I finally got round to addressing this. I think I succeeded but I don't really know what I'm doing with writing rust code.